### PR TITLE
feat: `createOktaOAuth2Client()`

### PR DESCRIPTION
**Pull request by jollytoad**
Okta provides you with your own domain, so you need to supply that to this provider via the OKTA_DOMAIN env var.

For more details see: https://developer.okta.com/docs/reference/api/oidc/#composing-your-base-url.

~~You can also optionally set OKTA_AUTHORIZATION_SERVER_ID too.~~

Fixes: #128

### CODE REVIEW

**Comment by jollytoad** [comment]
demo-screenshot

**Comment by jollytoad** [comment]
I think I've covered all the contributor requirements.
Thanks for such a useful lib.

**Comment by iuioiua** [comment]
I'll review within a day 👍🏾

File: src/providers/okta.ts [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+import type { WithRedirectUri, WithScope } from "./_types.ts";
+
+/**
+ * Creates an OAuth 2.0 client with Okta as the provider.
+ *
+ * Requires `--allow-env[=OKTA_CLIENT_ID,OKTA_CLIENT_SECRET,OKTA_DOMAIN]` permissions and environment variables:
+ * 1. `OKTA_CLIENT_ID`
+ * 2. `OKTA_CLIENT_SECRET`
+ * 3. `OKTA_DOMAIN`
+ *
```
Comment (iuioiua) [inline src/providers/okta.ts:13]:
```suggestion
 * 4. `OKTA_AUTHORIZATION_SERVER_ID` (if using a custom authorization server)
 *
```

File: src/providers/okta.ts [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+    & Partial<OAuth2ClientConfig>
+    & WithRedirectUri
+    & WithScope,
+): OAuth2Client {
+  const domain = Deno.env.get("OKTA_DOMAIN");
+  const authorizationServerId = Deno.env.get("OKTA_AUTHORIZATION_SERVER_ID");
+
+  const baseURL = `https://${domain}/oauth2/${
+    authorizationServerId ? `${authorizationServerId}` : ""
+  }`;
```
Comment (iuioiua) [inline src/providers/okta.ts:41]:
```suggestion
  const baseURL = `https://${domain}/oauth2/${authorizationServerId ?? ""}`;
```

File: src/providers/okta.ts [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+ * @see {@link https://developer.okta.com/docs/reference/api/oidc}
+ */
+export function createOktaOAuth2Client(
+  additionalOAuth2ClientConfig:
+    & Partial<OAuth2ClientConfig>
+    & WithRedirectUri
+    & WithScope,
+): OAuth2Client {
+  const domain = Deno.env.get("OKTA_DOMAIN");
+  const authorizationServerId = Deno.env.get("OKTA_AUTHORIZATION_SERVER_ID");
```
Comment (iuioiua) [inline src/providers/okta.ts:37]: Can you confirm this works with and without the `OKTA_AUTHORIZATION_SERVER_ID` environment variable set?

**Review by iuioiua — CHANGES_REQUESTED** [review]
Mostly LGTM! Thanks!

File: src/providers/okta.ts (hunk 2 above)
Comment (jollytoad) [reply src/providers/okta.ts:41]: Not sure how I missed this myself!

File: src/providers/okta.ts (hunk 3 above)
Comment (jollytoad) [reply src/providers/okta.ts]: I've removed `OKTA_AUTHORIZATION_SERVER_ID`, doesn't work, sorry, I shouldn't have included it in the PR.

Comment (iuioiua) [reply src/providers/okta.ts]: Oh. How come it doesn't work?

Comment (iuioiua) [reply src/providers/okta.ts]: I'll merge this PR now. Let's explore adding support for a custom authorization server in a separate issue.

**Review by iuioiua — APPROVED** [review]
LGTM! Thank you!

**Pull request merged**